app/web/pages/messages/[[...slug]].tsx

```diff
@@ -0,0 +1,62 @@
+import { appGetLayout } from "components/AppRoute";
+import GroupChatsTab from "features/messages/groupchats/GroupChatsTab";
+import GroupChatView from "features/messages/groupchats/GroupChatView";
+import MessagesHeader from "features/messages/MessagesHeader";
+import HostRequestView from "features/messages/requests/HostRequestView";
+import RequestsTab from "features/messages/requests/RequestsTab";
+import NotFoundPage from "features/NotFoundPage";
+import { GetStaticPaths, GetStaticProps } from "next";
+import { useRouter } from "next/router";
+import { serverSideTranslations } from "next-i18next/serverSideTranslations";
+import { ReactNode } from "react";
+import { messageTypeStrings } from "routes";
+
+export const getStaticPaths: GetStaticPaths = () => ({
+  paths: [],
+  fallback: "blocking",
+});
+
+export const getStaticProps: GetStaticProps = async ({ locale }) => ({
+  props: {
+    ...(await serverSideTranslations(locale ?? "en", ["global", "messages"])),
+  },
+});
+export default function LeaveReferencePage() {
+  const router = useRouter();
+  const slugs =
+    typeof router.query.slug === "undefined"
+      ? ["chats"]
+      : typeof router.query.slug === "string"
+      ? [router.query.slug]
+      : router.query.slug;
+  let content: ReactNode;
+
+  const tab = messageTypeStrings.find((valid) => valid === slugs?.[0]);
```

Comment:
```suggestion
  const tab = messageTypeStrings.find((valid) => valid === slugs[0]);
```

Reply:
It might not have any elements (`/messages/`

Reply:
Wouldn't `slugs` be an empty array in that case? I cloned and changed it and TS seems to think it's always an array. Is that not true?